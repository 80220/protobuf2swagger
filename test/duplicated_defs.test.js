const path = require('path');
const glob = require("glob");

const convert = require('../lib/convert');

const findAllProtoFiles = (rootDir) => {
  let files = glob.sync(path.resolve(__dirname, rootDir) + '/**/*');
  return files.filter((f)=>f.endsWith('.proto'));
}

describe('dup', () => {
  test('duplicated defs', async () => {
    const files = findAllProtoFiles('__fixtures__/duplicated_defs');
    console.log(files);
    const converted = await convert({
      files,
      customSchema: {
        swagger: '2.0',
      },
    });
    expect(converted).toMatchSnapshot();
  });
});


