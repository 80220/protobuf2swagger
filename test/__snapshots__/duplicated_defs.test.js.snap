// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dup duplicated defs 1`] = `
Object {
  "definitions": Object {
    "a.b.StaticRoute": Object {
      "properties": Object {
        "number": Object {
          "format": "int32",
          "type": "integer",
        },
      },
      "type": "object",
    },
    "a.b.c.d.CreateRouteRequest": Object {
      "properties": Object {
        "static_route": Object {
          "$ref": "#/definitions/a.b.c.d.StaticRoute",
        },
      },
      "type": "object",
    },
    "a.b.c.d.StaticRoute": Object {
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
  },
  "info": Object {
    "description": "",
    "title": "common",
    "version": "1",
  },
  "paths": Object {},
  "swagger": "2.0",
}
`;
